fn/async.go

```diff
@@ -0,0 +1,76 @@
+package fn
+
+import "errors"
+
+type Async[A any] interface {
```

Comment:
everything in this file that starts with a capital letter needs a godoc 🙏

Reply:
Why don't we have a linter for this since it seems a concise enough rule to state?